sdk/forms/src/main/java/com/klaviyo/forms/BridgeMessage.kt

```diff
@@ -0,0 +1,163 @@
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_AGGREGATE_EVENT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_PROFILE_EVENT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_DEEPLINK,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_CLOSE,
+                        IAF_TYPE_VERSION_KEY to 1
+                    ),
+                    mapOf(
+                        IAF_MESSAGE_TYPE_KEY to IAF_MESSAGE_TYPE_ABORT,
+                        IAF_TYPE_VERSION_KEY to 1
+                    )
+                )
+            ).toString()
+        }
+
+        /**
+         * Parse a native bridge message string into a [BridgeMessage]
+         *
+         * @throws IllegalStateException for unexpected message strings
+         */
+        fun decodeWebviewMessage(message: String): BridgeMessage {
+            val jsonMessage = JSONObject(message)
+            val jsonData = jsonMessage.optJSONObject(IAF_MESSAGE_DATA_KEY) ?: JSONObject()
+
+            return when (val type = jsonMessage.optString(IAF_MESSAGE_TYPE_KEY)) {
+                IAF_MESSAGE_HAND_SHOOK -> HandShook
+
+                IAF_MESSAGE_TYPE_SHOW -> Show(
+                    formId = jsonData.optString("formId").ifEmpty { "" }
+                )
+
+                IAF_MESSAGE_TYPE_AGGREGATE_EVENT -> AggregateEventTracked(
+                    payload = jsonData
+                )
+
+                IAF_MESSAGE_TYPE_PROFILE_EVENT -> ProfileEvent(
+                    event = Event(
+                        jsonData.getString("metric"),
+                        properties = jsonData.getEventProperties()
+                    )
+                )
+
+                IAF_MESSAGE_TYPE_DEEPLINK -> DeepLink(
+                    route = jsonData.getDeepLink()
+                )
+
+                IAF_MESSAGE_TYPE_CLOSE -> Close(
+                    formId = jsonData.optString("formId").ifEmpty { "" }
+                )
+
+                IAF_MESSAGE_TYPE_ABORT -> Abort(
+                    reason = jsonData.optString("reason").ifEmpty { "Unknown" }
+                )
+
+                else -> throw IllegalStateException("Unrecognized message type $type")
+            }
+        }
+
+        /**
+         * Parse [Event] properties for a [ProfileEvent] message
+         */
+        private fun JSONObject.getEventProperties(): Map<EventKey, Serializable> {
+            val map = mutableMapOf<EventKey, Serializable>()
+            val propertyMap = getJSONObject("properties")
+            propertyMap.keys().forEach {
+                try {
+                    map[EventKey.CUSTOM(it)] = propertyMap.get(it) as Serializable
```

Comment:
This is one small functional change that I made here: this was parsing only strings, we should be able to support any `Serializable` (and if it somehow isn't serializable, this will throw)